### PR TITLE
Rename Dispatcher invoke methods.

### LINKED ISSUE #816: Despatcher invoke methods need renaming.
The naming of `Dispatcher.InvokeTaskAsync` is confusing: the Async suffix usually means return a task. Remove this and rename `InvokeTaskAsync` as `InvokeAsync`. I think we only only need one? The task can be explicitly ignore if necessary.

### PR DESCRIPTION
**Pull request by grokys**
`InvokeAsync` -> `Post`
`InvokeTaskAsync` -> `InvokeAsync`

Fixes #816

### CODE REVIEW

File: src/Avalonia.Base/Threading/IDispatcher.cs [hunk 1]
```diff
@@ -34,6 +34,6 @@ public interface IDispatcher
         /// <param name="priority">The priority with which to invoke the method.</param>
         // TODO: The naming of this method is confusing: the Async suffix usually means return a task.
```
Comment (kekekeks) [inline src/Avalonia.Base/Threading/IDispatcher.cs]: Naming is fixed, right? )

**Review by kekekeks — APPROVED** [review]

**Comment by aelij** [comment]
I'm not too crazy about `Post` - it's an old word from the days of Win32 :)
Why do we need two methods anyway? You can always just ignore the `Task` return value (the TCS is created anyhow).

**Comment by grokys** [comment]
@aelij yeah I'm of the same opinion tbh. However in https://github.com/AvaloniaUI/Avalonia/issues/816#issuecomment-357425017, @kekekeks says that he'd like to keep both to reduce allocations. I'm not sure whether that's necessary personally though.

**Comment by aelij** [comment]
Hmm, how about returning something like `IDispatcherOperation` (implemented by `Job`) which could be awaitable (has a `GetAwaiter` method, where the task is lazily created)? The job is always allocated anyway.

**Comment by grokys** [comment]
I'd like to know if allocating an unused `TaskCompletionSource` is such a big deal? Do we even need to optimize for this case?

**Comment by aelij** [comment]
I guess it depends on the usage. How many invokes happen on average? Do these objects remain on gen0?

WPF always allocates a TCS BTW.

**Pull request merged**